frontend/static_vol/js/components/TournamentDetail.js

```diff
@@ -13,316 +15,292 @@ export default class TournamentDetail extends PageBase {
+        const blockTournamentRanking = document.querySelector('.blockTournamentRanking');
+        result.forEach(item => {
+            if (item.round !== undefined && !this.renderedRounds.has(item.round)) {
+                const roundList = this.generateRoundList(item);
+                if (roundList) {
+                    wrapTournamentRound.insertAdjacentHTML('afterbegin', roundList);
+                }
+                this.renderedRounds.add(item.round);
+            } else if (item.rankings !== undefined) {
+                blockTournamentRanking.innerHTML = this.generateRankingList(item.rankings);
+                blockTournamentRanking.classList.add('is-show');
+            }
+        });
+    }
+
+    generateRankingList(rankings) {
+        const avatar1 = this.avatarMap[rankings.winner_id] ? this.avatarMap[rankings.winner_id] : `/images/avatar_default.png`;
+        const avatar2 = this.avatarMap[rankings.second_id] ? this.avatarMap[rankings.second_id] : `/images/avatar_default.png`;
+        const avatar3 = this.avatarMap[rankings.third_id] ? this.avatarMap[rankings.third_id] : `/images/avatar_default.png`;
+        return `
+            <dl class="unitRanker">
+                <dt class="unitRanker_rank unitRanker_rank-1"><span>${labels.tournament.labelWinner}</span></dt>
+                <dd class="unitRanker_user">
+                    <img src="${avatar1}" alt="" width="50" height="50">
+                    ${rankings.winner}
+                </dd>
+            </dl>
+            <dl class="unitRanker">
+                <dt class="unitRanker_rank unitRanker_rank-2"><span>${labels.tournament.labelSecondPlace}</span></dt>
+                <dd class="unitRanker_user">
+                    <img src="${avatar2}" alt="" width="50" height="50">
+                    ${rankings.second}
+                </dd>
+            </dl>
+            <dl class="unitRanker">
+                <dt class="unitRanker_rank unitRanker_rank-3"><span>${labels.tournament.labelThirdPlace}</span></dt>
+                <dd class="unitRanker_user">
+                    <img src="${avatar3}" alt="" width="50" height="50">
+                    ${rankings.third}
+                </dd>
+            </dl>`;
+    }
+
+    generateMatch(match) {
+        let matchHtml = '';
+        const avatar1 = this.avatarMap[match.player1_id] ? this.avatarMap[match.player1_id] : `/images/avatar_default.png`;
+        const avatar2 = this.avatarMap[match.player2_id] ? this.avatarMap[match.player2_id] : `/images/avatar_default.png`;
+        matchHtml += `
+                <div class="blockMatch">
+                    <section class="blockMatch_player unitMatchPlayer">
+                        <header class="unitMatchPlayer_header">
+                            <img src="${avatar1}" alt="" width="50" height="50">
+                            <h4 class="unitMatchPlayer_title">${match.player1}</h4>
+                        </header>`;
+        if (match.score1) {
```

Comment:
```suggestion
        if (match.score1 !== undefined && match.score1 !== null) {
```
0点のときに「？」が表示されてしまうようです